### PR TITLE
Update google provider 4.x

### PR DESCRIPTION
**Pull request by thiesen**

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -2,6 +2,9 @@ terraform {
   required_version = ">= 0.14, < 2.0"
 
   required_providers {
-    google-beta = "~> 3.75"
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 3.75, < 5.0"
+    }
```
Comment (mariux) [inline versions.tf:8]: there are references in code to google-beta that need to be removed

**Review by mariux — CHANGES_REQUESTED** [review]

File: versions.tf (hunk 1 above)
Comment (mariux) [reply versions.tf:8]: unless it fails for still being beta

Comment (thiesen) [reply versions.tf:8]: > there are references in code to google-beta that need to be removed

hey @mariux, from https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository we get

> This resource is in beta, and should be used with the terraform-provider-google-beta provider. See Provider Versions for more details on beta resources.

and they state (https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_versions#using-the-google-beta-provider) that we should keep using the `google-beta` provider as in https://github.com/mineiros-io/terraform-google-artifact-registry-repository/blob/main/main.tf#L2

Comment (thiesen) [reply versions.tf]: oh I think I got it know :sweat_smile:

**Comment by thiesen** [comment]
waiting for release v0.0.2 of https://github.com/mineiros-io/terraform-google-artifact-registry-repository-iam to makr this PR as r4r

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**